dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/event/JdbcEventStore.java

```diff
@@ -466,6 +472,16 @@ public JdbcEventStore( StatementBuilder statementBuilder, JdbcTemplate jdbcTempl
                 event.getNotes().add( note );
                 notes.add( rowSet.getString( "psinote_id" ) );
             }
+
+            if ( !params.isSkipRelationship() )
```

Comment:
This still happens in a loop, so we get an extra N queries. If we extract 10000 events, thats 10000+ queries.

I can see this being solved in one of two ways:
1. include an aggregate of the relationships in the original SQL
2. gather all the event ids and do 1 SQL to get all relationships with those ids.

I think solution 1 is simpler than 2 tbh.

Reply:
it will be a problem if an event has more than one relationship.